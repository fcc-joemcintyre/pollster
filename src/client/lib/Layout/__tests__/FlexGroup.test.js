/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { FlexGroup } from '..';

test ('FlexGroup renders default layout', () => {
  const component = renderer.create (
    <FlexGroup>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexGroup>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('FlexGroup renders left justified component', () => {
  const component = renderer.create (
    <FlexGroup left>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexGroup>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('FlexGroup renders centered components', () => {
  const component = renderer.create (
    <FlexGroup center>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexGroup>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('FlexGroup renders right justfied components', () => {
  const component = renderer.create (
    <FlexGroup right>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexGroup>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('FlexGroup renders custom spaced components', () => {
  const component = renderer.create (
    <FlexGroup spacing='20px'>
      <button type='submit'>SUBMIT</button>
      <button type='button'>CANCEL</button>
      <button type='button'>CLEAR</button>
    </FlexGroup>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
