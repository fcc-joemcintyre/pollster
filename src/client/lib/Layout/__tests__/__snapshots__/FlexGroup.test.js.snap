// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexGroup renders centered components 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 -10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > * {
  margin: 10px 10px 0px 10px;
}

<div
  className="c0"
  spacing="10px"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;

exports[`FlexGroup renders custom spaced components 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 -20px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 > * {
  margin: 20px 20px 0px 20px;
}

<div
  className="c0"
  spacing="20px"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;

exports[`FlexGroup renders default layout 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 -10px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 > * {
  margin: 10px 10px 0px 10px;
}

<div
  className="c0"
  spacing="10px"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;

exports[`FlexGroup renders left justified component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 -10px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 > * {
  margin: 10px 10px 0px 10px;
}

<div
  className="c0"
  spacing="10px"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;

exports[`FlexGroup renders right justfied components 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 -10px;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 > * {
  margin: 10px 10px 0px 10px;
}

<div
  className="c0"
  spacing="10px"
>
  <button
    type="submit"
  >
    SUBMIT
  </button>
  <button
    type="button"
  >
    CANCEL
  </button>
  <button
    type="button"
  >
    CLEAR
  </button>
</div>
`;
